### PR TITLE
update security groups stage

### PR DESCRIPTION
**Pull request by anotherchrisberry**
This might be a little provider-specific, but there are worse stages in Orca.

@cfieber or @ajordens PTAL

### CODE REVIEW

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/UpdateLaunchConfigTask.groovy [hunk 1]
```diff
@@ -68,6 +68,7 @@ class UpdateLaunchConfigTask implements Task, DeploymentDetailsAware, CloudProvi
   private getAwsOps(Stage stage) {
     def operation = new HashMap(stage.context)
     operation.amiName = getImage(stage)
+    operation.asgName = operation.asgName ?: operation.serverGroupName
```
Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/UpdateLaunchConfigTask.groovy]: Do we still need `asgName`?

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/UpdateSecurityGroupsStage.groovy [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import com.netflix.spinnaker.orca.clouddriver.tasks.servergroup.UpdateLaunchConfigTask
+import com.netflix.spinnaker.orca.pipeline.LinearStage
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import groovy.util.logging.Slf4j
+import org.springframework.batch.core.Step
+import org.springframework.stereotype.Component
+
+@Slf4j
+@Component
+class UpdateSecurityGroupsStage extends LinearStage {
```
Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/UpdateSecurityGroupsStage.groovy]: I'm not sure this stage name really matches what it's doing.

`UpdateAsgSecurityGroupsStage`?

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/UpdateLaunchConfigTask.groovy (hunk 1 above)
Comment (anotherchrisberry) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/UpdateLaunchConfigTask.groovy]: Unfortunately, it's required by Clouddriver in the `updateAsgLaunchConfiguration` operation, and I didn't want to break anybody already using it.

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/UpdateSecurityGroupsStage.groovy (hunk 2 above)
Comment (anotherchrisberry) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/UpdateSecurityGroupsStage.groovy]: Sounds reasonable. I'll rename it to `UpdateSecurityGroupsForServerGroupStage`

**Pull request merged**